test/controllers/assignments_controller_test.rb

```diff
@@ -516,4 +516,45 @@ class AssignmentsControllerTest < ActionController::TestCase
 
     assert_redirected_to @cs101
   end
+
+  test "should allow for graders with files" do
+    sign_in @fred
+    upload_file = fixture_file_upload(
+      "files/HelloSingle/hello.c",'application/octet-stream')
+    assert_difference('Assignment.count') do
+      post :create, params: {
+             course_id: @cs101.id,
+             assignment: {
+               teamset_plan: "none",
+               assignment: "Dance a jig.",
+               points_available: 100,
+               name: "Useful Work",
+               due_date: '2019-05-22',
+               available: '2011-05-22',
+               type: "Files",
+               lateness_config_attributes: {
+                 "type"=>"LatePerHourConfig",
+                 "frequency"=>"1",
+                 "percent_off"=>"25",
+                 "max_penalty"=>"100",
+                 "days_per_assignment"=>"22"
+               },
+               graders_attributes: {
+                 "1477181088065"=> {
+                   "type"=>"JavaStyleGrader",
+                   "avail_score"=>"50",
+                   "order"=>"1",
+                   "_destroy"=>"false",
+                   "upload_file" => upload_file
+                 }
+               }
+             }
+           }
+    end
+
+    @assn = assigns(:assignment)
+    assert_equal @assn, @assn.graders.first.upload.assignment
```

Comment:
Also check that the upload's `upload_dir` starts with `Upload.base_upload_dir.join(@cs101.is.to_s, @assn.id.to_s)`, and that `File.exists?(upload.submission_path)` -- be a bit more thorough ensuring that everything went correctly